File: IDPSpin/NetworkBuffer.py
import threading
from threading import Semaphore

class NetworkBuffer(object):

    def __init__(self):
        self.Input = []
        self.Output = []
        self.InputSemaphore = threading.Semaphore(1)
        self.OutputSemaphore = threading.Semaphore(1)
        
    def Append(self, var):       
        self.InputSemaphore.acquire()
        self.Input.append(var)
        self.InputSemaphore.release()

    def Pop(self):
        self.InputSemaphore.acquire()
        
        output = ""
        try:
            output = self.Input.pop()
        except:
            output = ""
        self.InputSemaphore.release()
        return output

    

File: VisionEngine/VisionEngine/VisionEngine/Network/NetworkBuffer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace VisionEngine.Network
{
    class NetworkBuffer
    {
        private Stack<string> Buffer;
        private Semaphore BufferSemaphore;


        public NetworkBuffer()
        {
            Buffer = new Stack<string>();
            BufferSemaphore = new Semaphore(1, 1);
        }


        public void Append(string Data)
        {
            BufferSemaphore.WaitOne();
            Buffer.Push(Data);
            BufferSemaphore.Release();
        }


        public string Get()
        {
            BufferSemaphore.WaitOne();
            string Data = Buffer.Pop();
            BufferSemaphore.Release();
            return Data;
        }
    }

}
